File: TPS/TP01/Atv5/somaRec.c
#include <stdio.h>

int equal(char *str1, char *str2){						
	int i = 0;
	while(str1[i] != '\0' && str2[i] != '\0'){
		if(str1[i] != str2[i])// comparando strings se sao diferente
			return 0;
		i++;
	}
	//caso nao caia no if returna 1
	return 1;
}

void tirarN(char *s){
	int i = 0;

	while(s[i] != '\0'){
		if(s[i] == '\n'){ // função de tirar o \n da entrada do teclado
			s[i] = '\0';
			return;
		}
		i++;
	}
}

int somaRec(char *s){

	if(*s == '\0'){// condição de parada
		return 0;	
	}
	int resp = *s - '0'; // pega o valor ascii por exemplo 5 em ascii 
			     //é 53 e o 0 é 48 faz 53 - 48 = 5
			     //assim pegando valor exato;
	return resp + somaRec(s + 1);// chama recursiva fazendo 5 + a chamada da função novamente
}

int main(){
	
	char linha[100];
	
	scanf("%s", linha);
	tirarN(linha);
	while(!equal(linha,"FIM")){
		printf("%d\n",somaRec(linha));
		scanf("%s", linha);
		tirarN(linha);
	}
}

File: TPS/TP01/Atv12/SomaRec.java
import java.util.Scanner;

class SomaRec{

	public static int Soma(String s, int i, int tam){
		if(i >= tam){
			return 0;
		}
		char c = s.charAt(i);// Pega o caracter atual
		int soma = (c - '0');// faz a soma por exemplo 5 - 0 em ascii é 53 - 48 = 5
	
		return soma + Soma(s , i + 1, tam);// chamada recursiva
	}

	public static boolean equal(String str1, String str2){
		if(str1.length() != str2.length())
		       	// Verifica se as string tem tamanhos diferentes
			return false;

		for(int i = 0; i < str1.length(); i++){
			if(str1.charAt(i) != str2.charAt(i))
				//Verifica caracter por caracter
				//Se a String 1 é diferente de FIM retorna falso
				return false;
			}
		//Caso nao caia em nenhum if retorna verdadeiro
		return true;
	}

	public static void main(String[] args){
		Scanner scan = new Scanner(System.in);
		String s = scan.nextLine();

		while(!equal(s, "FIM")){
			int i = Soma(s, 0, s.length());
			MyIO.println(i);
			s = scan.nextLine();
		}
	}
}
